internal/build/lifecycle_execution_test.go

```diff
@@ -1399,6 +1399,106 @@ func testLifecycleExecution(t *testing.T, when spec.G, it spec.S) {
 				})
 			})
 		})
+
+		when("previous-image is used and builder is untrusted", func() {
+			when("image is invalid", func() {
+				it("errors", func() {
+					var imageName name.Tag
+					imageName, err := name.NewTag("/x/y/?!z", name.WeakValidation)
+					h.AssertError(t, err, "repository can only contain the runes `abcdefghijklmnopqrstuvwxyz0123456789_-./`")
+					lifecycle := newTestLifecycleExec(t, true, func(options *build.LifecycleOptions) {
+						options.Image = imageName
+						options.PreviousImage = "previous-image"
+					})
+					fakePhaseFactory := fakes.NewFakePhaseFactory()
+
+					err = lifecycle.Analyze(context.Background(), "test", "test", false, "", false, fakeCache, fakePhaseFactory)
+					h.AssertError(t, err, "invalid image name")
+				})
+			})
+
+			when("previous-image is invalid", func() {
+				it("errors", func() {
+					var imageName name.Tag
+					imageName, err := name.NewTag("/some/image", name.WeakValidation)
+					h.AssertNil(t, err)
+					lifecycle := newTestLifecycleExec(t, true, func(options *build.LifecycleOptions) {
+						options.PreviousImage = "%%%"
+						options.Image = imageName
+					})
+					fakePhaseFactory := fakes.NewFakePhaseFactory()
+					err = lifecycle.Analyze(context.Background(), "test", "test", false, "", false, fakeCache, fakePhaseFactory)
+					h.AssertError(t, err, "invalid previous image name")
+				})
+			})
+
+			when("--publish is false", func() {
+				it("successfully passes previous-image to analyzer", func() {
+					var imageName name.Tag
+					imageName, err := name.NewTag("/some/image", name.WeakValidation)
+					h.AssertNil(t, err)
+					lifecycle := newTestLifecycleExec(t, true, func(options *build.LifecycleOptions) {
+						options.PreviousImage = "previous-image"
+						options.Image = imageName
+					})
+					prevImage, err := name.ParseReference(lifecycle.Opts.PreviousImage, name.WeakValidation)
+					h.AssertNil(t, err)
+
+					fakePhaseFactory := fakes.NewFakePhaseFactory()
+					err = lifecycle.Analyze(context.Background(), "test", "test", false, "", false, fakeCache, fakePhaseFactory)
+					h.AssertNil(t, err)
+
+					lastCallIndex := len(fakePhaseFactory.NewCalledWithProvider) - 1
+					h.AssertNotEq(t, lastCallIndex, -1)
+
+					configProvider := fakePhaseFactory.NewCalledWithProvider[lastCallIndex]
+					h.AssertEq(t, configProvider.Name(), "analyzer")
+					h.AssertEq(t, lifecycle.Opts.Image.Name(), prevImage.Name())
```

Comment:
I'd like to consider alternatives to exposing a whole internal set of properties just to run an assertion on just one value. I don't have anything specific in mind but this PR shouldn't be as large as it is. Exposing an mutable property like this can easily become a slippery slope for some gnarly bugs.

Reply:
We already have getters for things like AppPath, AppDir, Builder. Adding another getter for `ImageName` sounds more reasonable to me.

Reply:
I'll look into it